logstash-core/src/main/java/org/logstash/instrument/metrics/MetricExt.java

```diff
@@ -140,30 +143,37 @@ protected IRubyObject getGauge(final ThreadContext context, final IRubyObject na
         );
     }
 
+    @Override
+    protected IRubyObject getTimer(final ThreadContext context,
+                                   final IRubyObject namespace,
+                                   final IRubyObject key) {
+        MetricExt.validateKey(context, null, key);
+        return collector.callMethod(context,
+                "get", new IRubyObject[]{namespace, key, TIMER}
+                );
+    }
+
     @Override
     protected IRubyObject doReportTime(final ThreadContext context, final IRubyObject namespace,
         final IRubyObject key, final IRubyObject duration) {
         MetricExt.validateKey(context, null, key);
-        return collector.callMethod(
-            context, "push", new IRubyObject[]{namespace, key, COUNTER, INCREMENT, duration}
-        );
+
+        final TimerMetric timer = timer(context, namespace, key).toJava(TimerMetric.class);
```

Comment:
The dance between `timer` and `getTimer` is hard to follow, but mirrors the `gauge`/`getGauge` dance that is done throughout. I am certain this can be made more efficient, but equally certain that making sense of it all is beyond the scope I can handle in a single PR.